llm/server.go

```diff
@@ -1498,7 +1502,7 @@ func (s *llmServer) Completion(ctx context.Context, req CompletionRequest, fn fu
 				return fmt.Errorf("error unmarshalling llm prediction response: %v", err)
 			}
 			switch {
-			case strings.TrimSpace(c.Content) == lastToken:
+			case lastToken != "" && (strings.TrimSpace(c.Content) == lastToken || strings.TrimSpace(c.Thinking) == lastToken):
```

Comment:
I don't think this is accurate since we would count a tokenRepeat if we had the same c.Content followed by the same c.Thinking token, even though the model produced a close thinking tag in between. Probably doesn't really matter in practice, but seems like it could lead to misunderstandings

Reply:
I worked through that mentally and since it would only be a single increment it shouldn't impact the sequence generation

Reply:
yeah that's what I meant by probably doesn't matter in practice.

But actually looking at it more closely, it's more broken than I thought since it now doesn't catch repetition in `c.Thinking`, only at the boundary between the two. This is a regression compared to the logic before that would catch repetition in any part of the message. Tbh this really belongs at a lower level now since the model could be repeating special characters or things the parser doesn't emit and could get stuck in an infinite loop (which I'm assuming what this relies on).

Reply:
Ah crap yes you're right. Let me rework this logic. thanks for the find

Reply:
no because `lastToken` isn't set for thinking in the default case (it'll be empty since content is empty, which I think is probably why you added that check at the beginning of this if-statement)